Validators/TriggerConditions.php

```diff
@@ -67,7 +67,7 @@ public function validate($value)
                 throw new Exception(Piwik::translate('TagManager_ErrorArrayMissingValue', array('comparison', $titlePlural, $index)));
             }
 
-            if (empty($condition['expected'])) {
+            if (!is_string($condition['expected']) || 0 === strlen($condition['expected'])) {
```

Comment:
@sgiehl this would still throw an exception for such case
https://3v4l.org/qkEcG

Reply:
```suggestion
            if (empty($condition['expected']) && strlen($condition['expected'])===0) {
```

Reply:
@AltamashShaikh But shouldn't the parameter actually always be a string, as it's coming from API?
Your suggestions actually wouldn't work either for `null` value, as PHP 8 would throw an error when passing `null` to `strlen`.

Reply:
yes it will fail for null values, I can see the values are always coming as string so it should be good to merge this change